### DISCUSSION TITLE
Some questions about cloud-hypervisor network

### DISCUSSION THREAD

**Discussion by ljrcore** [cloud-hypervisor/cloud-hypervisor#4890] [General]
Hi,

I read this file.
https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/io_throttling.md

I don't quite understand the bw_size and bw_refill_time here.
if the speed limit is 100Mbps, it is 12500000 bytes per second.
So if bw_refill_time = 100ms, is bw_size equal to 12500000/10 = 1250000?

Thanks

**Comment by rbradford** [comment]
If you want to rate limit to 100MBps then set the `bw_fill=12500000` and `bw_refill_time=1000`. So every second the budget is 12.5MiB which is 100Mb.

Of course you could refill more frequently e.g. make the `bw_refill_time` smaller but then you will need to adjust the `bw_fill` too.

However I think 1s refill time is as low as you would realistically want to go.

**Comment by ljrcore** [comment]
@rbradford Does cloud-hypervisor's QoS support whitelist or prioritization? For example, to target certain IP traffic to not pass through QoS or to make it higher priority?

Thanks

**Reply by rbradford** [reply]
I think you would use `iptables/nft` on the host to manage that.

**Reply by sboeuf** [reply]
Or something like tc filtering (traffic control).
